### PR TITLE
chore: bump to v0.11.0

### LINKED ISSUE #272: Can not preview latest Typst version (`v0.11.0`)
**Describe the bug**
On March 15, 2024, Typst `v0.11.0` was released, including many necessary changes to the `table` function, as well as introducing packages that are a huge improvement of QOL when using Typst.

Be that as it may, it is impossible to use any `v0.11.0` package and still be able to use `typst-preview` given that the following error is raised:

```bash
error: package requires typst 0.11.0 or newer (current version is 0.10.0)
  ┌─ /[...]/main.typ:2:8
  │
2 │ #import "@preview/drafting:0.2.0": *
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^
```

**To Reproduce**
Steps to reproduce the behavior (Library test):
1. Open a typst document.
2. Import any Typst package that requires `v0.11.0`
3. The error is raised.

**Expected behavior**
Preview the document with the new package's functionality.

**Package/Software version:**

VSCode version (Help → About):
```plain
Version: 1.87.2
Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
Date: 2024-03-08T15:20:57.520Z (1 wk ago)
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.1.0
```

typst-preview extension version: `v0.10.10`

**Logs:**

typst-preview server log(Output Panel -> typst-preview):
```bash
[2024-03-17T17:46:52Z DEBUG typst_ts_compiler::service::compile] CompileActor: memory event incoming
[2024-03-17T17:46:52Z DEBUG typst_ts_compiler::service::compile] CompileActor: fs event incoming Some(Update(FileChangeSet { removes: [], inserts: [("/[...]/main.typ", FileSnapshot { mtime: SystemTime { tv_sec: 1710697612, tv_nsec: 697234816 }, content: FileContent { len: 593 } })] }))
[2024-03-17T17:46:52Z DEBUG typst_ts_compiler::service::compile] CompileActor: fs event incoming Some(UpstreamUpdate { changeset: FileChangeSet { removes: [], inserts: [] }, upstream_event: Some(UpstreamUpdateEvent { invalidates: ["/[...]/main.typ"], opaque: Any { .. } }) })
[2024-03-17T17:46:52Z INFO  typst_ts_compiler::service::diag::console] /main.typ: compiling ...
    
[2024-03-17T17:46:52Z INFO  typst_ts_compiler::service::diag::console] /main.typ: Compilation failed after 222µs
    
error: package requires typst 0.11.0 or newer (current version is 0.10.0)
  ┌─ /[...]/main.typ:2:8
  │
2 │ #import "@preview/drafting:0.2.0": *
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Additional context**
First, I'd like to thank you for your amazing work in this extension which has done wonders for me and allowed me to improve my workflow a lot, and I truly appreciate the effort you've put into developing such an amazing tool.

I know this is probably due to the latest version of Typst (v0.11.0) not being fully supported yet, but, I was wondering, is there any possible workaround or alternative approach that would allow me to use Typst v0.11.0 and still benefit from the preview functionality?

In my current project, using the latest `v0.11.0` features is crucial due to the specific requirements of a somewhat important document I'm working on.

Regardless, thank you and sorry in advance.

### PR DESCRIPTION
**Pull request by Enter-tainer**
fix #272

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -112,7 +112,10 @@ impl CompilationHandle for CompilationHandleImpl {
 }
 
 /// If this file is not found, please refer to https://enter-tainer.github.io/typst-preview/dev.html to build the frontend.
+#[cfg(feature = "embed-html")]
 const HTML: &str = include_str!("../addons/vscode/out/frontend/index.html");
+#[cfg(not(feature = "embed-html"))]
+const HTML: &str = "HTML not embedded.";
```
Comment (Myriad-Dreamin) [inline src/lib.rs:118]: This is not a valid HTML string.

File: src/main.rs [hunk 2]
```diff
@@ -110,7 +110,7 @@ async fn main() {
         let world = TypstSystemWorld::new(CompileOpts {
             root_dir: root.clone(),
             font_paths: arguments.font_paths.clone(),
-            with_embedded_fonts: get_embedded_fonts(),
+            with_embedded_fonts: typst_assets::fonts().map(Cow::Borrowed).collect(),
```
Comment (Myriad-Dreamin) [inline src/main.rs:113]: `embed-fonts` is not valid anymore if you do that. People will always have to run preview with embedded fonts. Either remove feature `embed-fonts = ["typst-assets/fonts"]` and always enable "typst-assets/fonts", or revert this change.

Comment (Myriad-Dreamin) [reply src/main.rs:113]: I think I may mistakely think of it. there is no problem.

Comment (Enter-tainer) [reply src/main.rs:113]: https://github.com/typst/typst-assets/blob/main/src/lib.rs#L65

File: src/lib.rs (hunk 1 above)
Comment (Enter-tainer) [reply src/lib.rs:118]: I think that doesn't really matter, as long as it can display something strange in browser
